Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM golang:1.17-buster@sha256:3e663ba6af8281b04975b0a34a14d538cdd7d284213f83f05aaf596b80a8c725 as builder
```

Comment:
Why is this Dockerfile required?

Reply:
In this Dockerfile, compiling the binary inside the docker container and copy into host machine. 

To compile this, we need Golang 1.16+ . I have older golang version for my project requirement. Don't want to upgrade golang for this. That's why taken this approach. Thought of everyone will face similar issue while contributing. So that no need to mess up the developer development machine. 

I will remove this If you strongly believe this is not needed.

Reply:
I see, not unreasonable. What are your thoughts if we call this `Dockerfile.local` and change the wording in the Readme from s/Build/Development.

.. and then update the Readme wording to indicate this can be used to build local `goose` binaries without having the latest Go version installed locally?

I think we'll eventually want to add a proper Dockerfile to the project that we publish to a registry, so I'd like to reserve the `Dockerfile` for that.

Reply:
Done. 

Update the README.md and rename the Dockerfile to Dockerfile.local